canonical-web-and-design/maas-ui
# feat(ui): build Map Subnet form in Subnet details page.

## Done

- Build "Map subnet" form in Subnet details page

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the details page of an IPv4 subnet
- Choose "Map subnet" from the take action menu and check that you can successfully submit the form (if the subnet is not attached to any rack controller it will error, otherwise it should succeed. We only get that info from `ControllerDetails` so we can't validate that on the frontend with the current API)
- Go to the details page of an IPv6 subnet and select "Map subnet"
- The form should be disabled with an error message

## Fixes

Fixes canonical-web-and-design/app-tribe#651
